jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -261,6 +284,27 @@ def _jwst2data(file_obj, ext, data_label):
 
     return data, new_data_label
 
+# ---- Functions that handle input from Roman ASDF files -----
+
+
+def _roman_2d_asdf_to_glue_data(file_obj, data_label):
+    ext_list = ['data', 'dq', 'err', 'var_poisson', 'var_rnoise']
+
+    for ext in ext_list:
+        comp_label = ext.lower()
+        new_data_label = f'{data_label}[{comp_label}]'
+        data = Data(label=new_data_label)
+
+        # For now, set bunit to a blank string.
+        bunit = ''
+        component = Component.autotyped(np.array(getattr(file_obj, ext)), units=bunit)
+        data.add_component(component=component, label=comp_label)
+        data.coords = getattr(getattr(file_obj, 'meta'), 'wcs')
```

Comment:
Could be optimized, but you get the idea. We can pass in the metadata too.
```suggestion
        data.coords = getattr(getattr(file_obj, 'meta'), 'wcs')
        data.meta.update(dict(getattr(file_obj, 'meta')))
```